### DISCUSSION TITLE
I made a pixel art to text art conversion tool, motivated by arttime

### DISCUSSION THREAD

**Discussion by Nellousan** [poetaman/arttime#47] [Show and tell]
Hello ! 
I made a pixel art to text art converter using 24bit ANSI Escape codes for colors.
It was motivated by the discovery of arttime, after seeing the `taco` art, i did not even know it was possible to make pixel art this clear in the terminal, it made me want to use my own pixel arts with arttime. 
I didn't find any tool that did what i was looking for so it was an opportunity to make it myself.
here is what it looks like:
![](https://images2.imgbox.com/9a/a4/kZDPFR3O_o.png)
The converter translate the pixels 1:1 and does not make any modification to the original image.
here is the [link](https://github.com/Nellousan/px2ansi) to the repository, if interested.

**Comment by poetaman** [comment]
@Nellousan Awesome, that looks great! Couple of questions: 1) Is it correct understanding that 2 horizontal and two vertical pixels translate to one text character? I would guess so having seen the glyphs of Unicode's "Symbols for Legacy Computing" area. 2) Would you know of a place to find quality pixel art, something that fits in under 80 character width and 30 character height? We can take permission from their creators to convert some of it using px2ansi and add to arttime's repository.

**Reply by Nellousan** [reply]
Thanks !

1) I'm making use of the upper and lower half block unicode characters (ie `▀` and `▄`), on most terminals they fill almost perfectly half of the cell by default, dividing it in 2 almost perfect squares (ie pixels). So 1 text character is 2 vertical pixels.

2) I think the [r/PixelArt](https://www.reddit.com/r/PixelArt/) on reddit is a great place to find good, relatively low resolution pixel arts such as [this one](https://www.reddit.com/r/PixelArt/comments/zjlre4/spring/), but finding one under 80x30 specifically that is not just a sprite form someone's game is pretty rare, i would say. Other than that i don't know any niche place to find low res arts specifically, but i found some places that could be worth to checkout: [Pixilart](https://www.pixilart.com/), [Pixeljoint](https://pixeljoint.com/), and [a list](https://lospec.com/articles/pixel-art-communities/) of px art communities.

**Reply by poetaman** [reply]
@Nellousan Ah I see, will check those links! I think "symbols for legacy computing" has all the glyphs that divide a character in 2 vertical pixels and 3 horizontal ones. A font that is rectangular would mean that pixels are mostly square. IMO it will make it possible to display larger pieces in 80x30 characters (essentially art of size 160*90 pixels, which is also a nice 16:9 ratio coincidentally). Given there are 6 pixels, we need 2^6 (=64) glyphs. Glyphs 1FB00-1FB3B are 60 of them + 2 Unicode Right/Left half block (U+2590, U+258C) + 1 space (U+0020) + 1 full block (U+2588) = all 64 patterns.

![Symbols_for_Legacy_Computing_Unicode_block](https://user-images.githubusercontent.com/71736629/206981376-998c42ae-7390-49b7-a14c-1d7eafcd9eb3.png)

**Reply by Nellousan** [reply]
I absolutely agree that those characters would be the way to increase the resolution. The main issue is with colors, you can only have 2 colors per cell: foreground and background, so it is not possible to have an exact 1:1 color reproduction of any pixel arts with this because you would have to pick 2 colors for 6 "pixels". It is absolutely doable, but would alter the original work colors. The colors would not be an issue on monochrome arts tho.\
Also, as you mentioned, these appear mostly squared on the terminal but they are still 2:3 little rectangles so it would stretch the original image. Characters that divide the cell in 2x4 blocks would be perfect but i'm not aware of such chars existing.

Actually I was planning on maybe working on these kind of features later and turning the project into a real one if it gets people interested. Right now it's a fancy project with a fancy readme, but the actual conversion script is very simple and fits on around 50 lines.

**Reply by poetaman** [reply]
Yes that makes sense. Btw, I think there are new escape sequences that might interest you if/when you decide to work more on it. It can fit more colors and pixels in a character on terminal. Not sure how many terminals support the feature though. https://gitlab.freedesktop.org/terminal-wg/specifications/-/issues/23
